tests/test_rigid_physics.py

```diff
@@ -2352,6 +2352,62 @@ def test_drone_advanced(show_viewer):
     assert abs(quat_1[2] - quat_2[2]) < tol
 
 
+@pytest.mark.required
+@pytest.mark.parametrize("backend", [gs.cpu])
+def test_get_weld_constraints_basic(show_viewer, tol):
+    scene = gs.Scene(
+        sim_options=gs.options.SimOptions(gravity=(0.0, 0.0, 0.0)),
+        show_viewer=show_viewer,
+    )
+    cube1 = scene.add_entity(gs.morphs.Box(size=(0.05,) * 3, pos=(0.0, 0.0, 0.05)))
+    cube2 = scene.add_entity(gs.morphs.Box(size=(0.05,) * 3, pos=(0.2, 0.0, 0.05)))
+    scene.build(n_envs=1)
+
+    link_a = torch.tensor([cube1.base_link.idx], dtype=gs.tc_int, device=gs.device)
+    link_b = torch.tensor([cube2.base_link.idx], dtype=gs.tc_int, device=gs.device)
+
+    scene.sim.rigid_solver.add_weld_constraint(link_a, link_b)
+    scene.step()
+
+    welds = scene.sim.rigid_solver.get_weld_constraints(as_tensor=True, to_torch=False)
+    env_id = 0 if "env" not in welds else int(welds["env"][0])
+
+    row = np.array([env_id, int(welds["obj_a"][0]), int(welds["obj_b"][0])], dtype=np.int32)
+    ref = np.array([0, link_a.item(), link_b.item()], dtype=np.int32)
+    assert_allclose(row, ref, tol=tol)
+
+    cube1.set_dofs_velocity(
+        velocity=np.array([1.0, 0.0, 0.0], dtype=np.float32),
+        dofs_idx_local=np.array([0, 1, 2], dtype=np.int32),
+        envs_idx=None,
+    )
+    for _ in range(200):
+        scene.step()
+    assert_allclose(cube1.get_vel(), cube2.get_vel(), tol=tol)
```

Comment:
What is the point of this ? You are only validating API, not physics. This is done in another test? Just remove it.